tools/scripts/markdownlint.sh

```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
+
+# Copyright (c) 2023 by Apex.AI Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+set -e
+
+SCOPE=${1:-full} # Can be either `full` for all files or `hook` for formatting with git hooks
+
+
+fail() {
+    printf "\033[1;31merror: %s: %s\033[0m\n" ${FUNCNAME[1]} "${1:-"Unknown error"}"
+    exit 1
+}
+
+hash git || fail "git not found"
+
+# Check if we have at least a specific markdownlint version installed
+MARKDOWNLINT_VERSION=0.33.0
+MARKDOWNLINT_CMD="markdownlint"
+if ! command -v $MARKDOWNLINT_CMD &> /dev/null
+then
+    MARKDOWNLINT_MAJOR_VERSION=$(markdownlint --version | sed -rn 's/.*([0-9][0-9])\.[0-9].*/\1/p')
+    if [[ $MARKDOWNLINT_MAJOR_VERSION -lt "$MARKDOWNLINT_VERSION" ]]; then
+        echo "Warning: markdownlint version $MARKDOWNLINT_VERSION or higher is not installed."
+        echo "Code will not be formatted."
+        exit 0
+    else
+        MARKDOWNLINT_CMD="markdownlint"
+    fi
+fi
+
+cd "$(git rev-parse --show-toplevel)"
+
+if [[ "$SCOPE" == "hook"* ]]; then
+    md_files=$(git diff --cached --name-only --diff-filter=ACMRT | grep -E "\.md"| cat)
```

Comment:
I think it's good to run the linter on all markdown files.

@elfenpiff What do you think?

Reply:
Yes, it is actually very important!

Just think file A links to file B and one moves file B. Then the link of file A is broken but this will be never detected since only the modified files are checked and A was never modified.